subprojects/language-groovy/src/main/java/org/gradle/api/tasks/compile/GroovyCompile.java

```diff
@@ -60,24 +64,58 @@
 public class GroovyCompile extends AbstractCompile {
     private Compiler<GroovyJavaJointCompileSpec> compiler;
     private FileCollection groovyClasspath;
+    private ConfigurableFileCollection compilerPluginClasspath;
     private final CompileOptions compileOptions;
     private final GroovyCompileOptions groovyCompileOptions = new GroovyCompileOptions();
 
     public GroovyCompile() {
         CompileOptions compileOptions = getServices().get(ObjectFactory.class).newInstance(CompileOptions.class);
         this.compileOptions = compileOptions;
+        this.compilerPluginClasspath = getServices().get(ObjectFactory.class).fileCollection();
         CompilerForkUtils.doNotCacheIfForkingViaExecutable(compileOptions, getOutputs());
     }
 
+    @Override
+    @CompileClasspath
+    public FileCollection getClasspath() {
+        // Note that this is an approximation and must be fixed before de-incubating getCompilerPluginClasspath()
+        // See https://github.com/gradle/gradle/pull/9513
+        return super.getClasspath();
+    }
+
+    /**
+     * The classpath for compiler plugin to run, for example, AST transformation.
+     *
+     * @since 5.6
+     */
+    @Classpath
+    @Incubating
+    public FileCollection getCompilerPluginClasspath() {
```

Comment:
The return type needs to be `ConfigurableFileCollection`, if it isn't you cannot configure it from Java code.